lib/lita/external/robot.rb

```diff
@@ -1,9 +1,26 @@
+require 'zk'
+
 module Lita
   module External
     class Robot < ::Lita::Robot
+
+      def initialize_leader_lock
+        return unless ENV['ZOOKEEPER_PEERS']
+        zk_peers = ENV['ZOOKEEPER_PEERS']
+        zk_lock = "lita_#{config.robot.name}_#{ENV['ENV']}"
+        @locker = ZK::Locker.exclusive_locker(ZK.new(zk_peers), zk_lock)
+        @locker.lock(wait: false)
+      end
```

Comment:
Agreed, I was originally passing down more state from the caller, no need for separate method now.  Is there a better alternative to the `ENV` settings?